clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -298,10 +303,13 @@ private void autoCommitSyncNowWithRetries(OffsetCommitRequestState requestAttemp
             if (error == null) {
                 result.complete(null);
             } else {
-                if (error instanceof RetriableException || isStaleEpochErrorAndValidEpochAvailable(error)) {
+                if ((error instanceof RetriableException || isStaleEpochErrorAndValidEpochAvailable(error))) {
                     if (error instanceof TimeoutException && requestAttempt.isExpired) {
                         log.debug("Auto-commit sync timed out and won't be retried anymore");
                         result.completeExceptionally(error);
+                    } else if (error instanceof UnknownTopicOrPartitionException) {
+                        log.debug("Auto-commit sync failed because topic or partition were deleted");
```

Comment:
nit: would probably make the log clearer to all if we add the before revocation part `Auto-commit sync **before revocation** failed` (just like you did for the func name, like it)

Reply:
Good idea!

Reply:
I added the `before revocation` to all log messages in this method, since it actually applies to all.